### ISSUE TITLE
Dock only hides on half of screen

### ISSUE THREAD

**Issue by fliberdygibits** [micheleg/dash-to-dock#2328]
![Screenshot From 2024-11-07 18-05-16](https://github.com/user-attachments/assets/0827f31e-8de2-43c1-846c-285acb9c6f12)
![Screenshot From 2024-11-07 18-05-07](https://github.com/user-attachments/assets/9f732849-06f0-4727-ae8c-e394344d3281)

Per the screenshots here the dock only intelligently hides when I maximize a window to the left.  When I do so on the right it does not hide.  This is with intelligent autohide set on as well as dodge windows.  

This is on EndeavourOS with Kernel 6.11.6 and Gnome 47.1

**Comment by Hillard28** [comment]
Yeah, I've noticed this issue. Definitely bugs me as I like the intelligent autohide, but until then just have to set it to hide by default.

**Comment by adhadse** [comment]
For now, I've set the "Dodge windows" to "All Windows" in "Intelligent Autohide"

**Comment by Hillard28** [comment]
Oh, interesting! I was under the impression that "All Windows" didn't work as well, but it appears to.

**Comment by Hillard28** [comment]
Super weird. Both settings seem to work, but only so long as the Dash to Dock settings are open. Once I close them the right-side autohide issue returns.

**Comment by sergio-costas** [comment]
Honestly, this really sounds like a bug in gnome shell, specifically in the part that returns the windows position... I'll think on a test for you to run...

**Comment by Hillard28** [comment]
Even though it appears to work properly while the dash-to-dock settings are open? Let me see if I can get a video of what I mean.

**Comment by vanvugt** [comment]
Looks like this is triggered by the display scale being higher than 100%?

**Comment by giacof0** [comment]
Hello, I'm the author of #2303, which reports the same bug as this one. I'd like to update whoever is subscribed to this issue by saying that this problems only presents itself with blur-my-shell enabled regardless of other previously suspected culprits like display scale or odd aspect ratios. I don't know whether the issue lies in dash-to-dock or blur-my-shell. Read the comments in #2303 for more details.

**Comment by sergio-costas** [comment]
I see in the blur-my-shell github repo that it has specific code for Dash-to-Dock. Can you try to disable it from the extension settings, to see if it is that specific code, please?

**Comment by Hillard28** [comment]
Argh, I should have checked with all other extensions disabled, sorry. I wasn't even thinking about the BMS dock effects.

**Comment by vanvugt** [comment]
Duplicate of #2303
